catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPluginImpl.java

```diff
@@ -0,0 +1,170 @@
+        removeFromTable = b;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean addActiveSearch(ActiveSearch as) {
+        if (as == null) {
+            LOGGER.warn("Cannot add null ActiveSearch to table!");
+            return false;
+        }
+        activeSearchTable.put(as.getUniqueID(), as);
+        return true;
+    }
+
+    public QueryMonitorPluginImpl() {
+        LOGGER.info("QueryMonitor started");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean removeActiveSearch(UUID uniqueID) {
+        if (uniqueID == null) {
+            LOGGER.warn("Can't remove active search with null ID.");
+            return false;
+        }
+        if (activeSearchTable == null) {
+            LOGGER.warn("Can't remove active search. Search not found in table.");
+            return false;
+        }
+        if (activeSearchTable.remove(uniqueID) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method that is implemented for {@link PreFederatedQueryPlugin}. Uses the given {@link Source}
+     * and {@link QueryRequest} information to create a new {@link ActiveSearch} to add to the ActiveSearchTable.
+     *
+     * @param source {@link Source} that corresponds to source the search is querying
+     * @param input  {@link QueryRequest} that corresponds to request generated when a user queried the source
+     * @return {@link QueryRequest} that was given as a parameter with updated property information corresponding
+     * to the {@link ActiveSearch}'s {@link UUID}
+     */
+
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        try {
+            if (source == null) {
+                LOGGER.warn("Source given is null.");
+            }
+            UUID uniqueID = UUID.randomUUID();
+            String cqlQuery = null;
+            if (input == null) {
+                LOGGER.warn("QueryRequest given to queryMonitor was null!");
+            } else {
+
+                cqlQuery = input.toString();
+                if (cqlQuery.equals("unknownCQLQuery")) {
+                    LOGGER.warn("CQL Query not set as expected in QueryRequest");
+                } else {
+                    int filterStartIndex = cqlQuery.indexOf("filter=[");
+                    int endOfQueryIndex = cqlQuery.indexOf(" ],startIndex");
+                    if (filterStartIndex == -1 || endOfQueryIndex == -1) {
+                        LOGGER.warn(
+                                "Unable to determine start and end of query terms. Using entire queryRequest instead.");
+                    } else {
+                        cqlQuery = cqlQuery.substring(filterStartIndex + 9, endOfQueryIndex);
+                    }
+                }
+                input.getProperties()
+                        .put("GUID", uniqueID);
```

Comment:
Since this is a key you use multiple places in the file it would be easier to maintain if you make this a private static final field (`ACTIVE_SEARCH_GUID_KEY` or something mebbe?)